### ISSUE TITLE
Getting warning while training with transformer/biencoder

### ISSUE THREAD

**Issue by coderkhaleesi** [facebookresearch/ParlAI#2245]
**Bug description**

I am getting a warning when trying to run this command:
python3 -u examples/train_model.py     --init-model zoo:pretrained_transformers/bi_model_huge_reddit/model     --batchsize 2 -pyt convai2     --shuffle true --model transformer/biencoder --eval-batchsize 6     --warmup_updates 100 --lr-scheduler-patience 0     --lr-scheduler-decay 0.4 -lr 5e-05 --data-parallel True     --history-size 20 --label-truncate 72 --text-truncate 360     --num-epochs 10.0 --max_train_time 200000 -veps 0.5 -vme 8000     --validation-metric accuracy --validation-metric-mode max     --save-after-valid True --log_every_n_secs 20 --candidates batch     --dict-tokenizer bpe --dict-lower True --optimizer adamax     --output-scaling 0.06      --variant xlm --reduction-type mean --share-encoders False      --learn-positional-embeddings True --n-layers 12 --n-heads 12      --ffn-size 3072 --attention-dropout 0.1 --relu-dropout 0.0 --dropout 0.1      --n-positions 1024 --embedding-size 768 --activation gelu      --embeddings-scale False --n-segments 2 --learn-embeddings True      --share-word-embeddings False --dict-endtoken start --fp16 True   --model-file /tmp/tmppolymodel -bs 2

**Reproduction steps**

Just run the above command

**Expected behavior**

I expected the warning to be not there. And is there is a reason, I don't understand it.

**Logs**

***************************************************************************
[ WARNING ] : your model is being loaded with opts that do not exist in the model you are initializing the weights with: evaltask: None,eval_batchsize: 6,display_examples: False,num_epochs: 10.0,max_train_time: 200000.0,validation_every_n_secs: -1,save_every_n_secs: -1,save_after_valid: True,validation_every_n_epochs: 0.5,validation_max_exs: 8000,short_final_eval: False,validation_patience: 10,validation_metric: accuracy,validation_metric_mode: max,validation_cutoff: 1.0,load_from_checkpoint: False,validation_share_agent: False,aggregate_micro: False,metrics: default,tensorboard_log: False,pytorch_teacher_task: convai2,pytorch_teacher_dataset: None,pytorch_datapath: None,numworkers: 4,pytorch_preprocess: False,pytorch_teacher_batch_sort: False,batch_sort_cache_type: pop,batch_length_range: 5,shuffle: True,batch_sort_field: text,pytorch_context_length: -1,pytorch_include_labels: True,dict_maxexs: -1,dict_include_valid: False,dict_include_test: False,log_every_n_secs: 20.0,interactive_mode: False,adam_eps: 1e-08,encode_candidate_vecs_batchsize: 256,rank_top_k: -1,inference: max,topk: 5

***************************************************************************
[ WARNING ] : your model is being loaded with opts that differ from the model you are initializing the weights with. Add the following args to your run command to change this: 

--task convai2 --download-path /private/home/edinan/ParlAI/downloads --batchsize 512 --datapath /private/home/edinan/ParlAI/data --single-turn False --rank-candidates True --candidates inline --encode-candidate-vecs False --parlai-home /private/home/edinan/ParlAI
***************************************************************************
Dictionary: loading dictionary from /home/ubuntu/ParlAI/data/models/pretrained_transformers/bi_model_huge_reddit/model.dict
[ num words =  54944 ]
Total parameters: 256081920
Trainable parameters:  256081920
Loading existing model parameters from /home/ubuntu/ParlAI/data/models/pretrained_transformers/bi_model_huge_reddit/model
[creating task(s): pytorch_teacher]
[ dictionary already built .]
[ no model with opt yet at: /tmp/tmppolymodel(.opt) ]
```
Output goes here
```

**Additional context**

**Comment by stephenroller** [comment]
These are there to warn you that there are some option mismatches. We use that warning to help us identify mismatched options that might be causing problems when resuming training. A quick glance over the options doesn't set off any alarm bells for me, so I think you don't need to worry, as long as training is happening fine.

If you think the wording could be improved, I'm very open to a PR improving this.

On our roadmap is explicit annotation of which options are safe to change and which are not (#2177). Once that annotation exists, we can start to suppress the warning for many of those options.
